name.abuchen.portfolio.tests/src/name/abuchen/portfolio/datatransfer/pdf/baaderbank/BaaderBankPDFExtractorTest.java

```diff
@@ -67,27 +67,29 @@
 import name.abuchen.portfolio.money.CurrencyUnit;
 import name.abuchen.portfolio.money.Money;
 import name.abuchen.portfolio.money.Values;
+import name.abuchen.portfolio.online.SecuritySearchProvider;
 import name.abuchen.portfolio.online.impl.CoinGeckoQuoteFeed;
+import name.abuchen.portfolio.online.impl.CoinGeckoQuoteFeed.Coin;
 
 @SuppressWarnings("nls")
 public class BaaderBankPDFExtractorTest
 {
     BaaderBankPDFExtractor extractor = new BaaderBankPDFExtractor(new Client())
     {
         @Override
-        protected CoinGeckoQuoteFeed lookupFeed()
+        protected List<SecuritySearchProvider> lookupCryptoProvider()
         {
             // mock the list of coins to avoid remote call
-            return new CoinGeckoQuoteFeed()
+            return List.of(new SecuritySearchProvider()
             {
                 @Override
-                public synchronized List<Coin> getCoins() throws IOException
+                public List<ResultItem> getCoins() throws IOException
                 {
                     return List.of( //
-                                    new Coin("bitcoin", "BTC", "Bitcoin"), //
-                                    new Coin("ethereum", "ETH", "Ethereum"));
+                                    new Coin("bitcoin", "BTC", "Bitcoin").asResultItem(), //
+                                    new Coin("ethereum", "ETH", "Ethereum").asResultItem());
                 }
-            };
+            });
         }
```

Comment:
Can we look into making this a global list of coins tested so far?
Maybe outsource the list?

Reply:
good Idea. I have extracted into a separate helper class with a static method.

Unfortunately, it is a method of the AbstractPDFExtractor which means we still have to override the method in each test.